spec/webauthn/authenticator_assertion_response_spec.rb

```diff
@@ -5,35 +5,40 @@
 RSpec.describe WebAuthn::AuthenticatorAssertionResponse do
   let(:challenge) { fake_challenge }
   let(:encoded_challenge) { WebAuthn::Utils.ua_encode(challenge) }
+  let(:original_origin) { fake_origin }
 
   it "is valid if everything's in place" do
     assertion_response = WebAuthn::AuthenticatorAssertionResponse.new(
-      client_data_json: encoded_fake_client_data_json(challenge: challenge, type: "webauthn.get"),
+      client_data_json: encoded_fake_client_data_json(challenge: challenge,
+                                                      origin: original_origin,
+                                                      type: "webauthn.get"),
       authenticator_data: fake_authenticator_data
     )
 
-    expect(assertion_response.valid?(encoded_challenge)).to be_truthy
+    expect(assertion_response.valid?(encoded_challenge, original_origin)).to be_truthy
   end
 
   describe "type validation" do
     it "is invalid if type is not get" do
       assertion_response = WebAuthn::AuthenticatorAssertionResponse.new(
-        client_data_json: encoded_fake_client_data_json(challenge: challenge, type: "webauthn.create"),
+        client_data_json: encoded_fake_client_data_json(challenge: challenge,
+                                                        type: "webauthn.create"),
```

Comment:
Passing `original_origin` to `encoded_fake_client_data_json`? So that we are explicit that the client data has the same origin as the one we are passing in the `valid?`.

Reply:
We could do that, I don't mind it :). But my reasoning was that it's not something I'm testing specifically in the spec, then I don't mind to be "hidden".

Reply:
I strongly agree with you that hiding (or encapsulating) something in test-helpers that's not the actual testing target is a good thing.

That said, I am not sure it's what actually happening in this test case. It seems to me that is partially hidden.

The fact that we are passing `original_origin` in `expect(assertion_response.valid?(encoded_challenge, original_origin)).to be_falsy` means that, instead of "origin validation" being totally encapsulated inside test-helpers, the test-case ends-up with a dependency on the exact value the `fake_client_data_json` helper is defaulting `origin` to.

In other words, if `fake_client_data_json` changes, and origin defaults to something different, say

```diff
diff --git a/spec/spec_helper.rb b/spec/spec_helper.rb
index a790ccf..fc5c702 100644
--- a/spec/spec_helper.rb
+++ b/spec/spec_helper.rb
@@ -70,7 +70,7 @@ end
 def fake_client_data_json(challenge: nil, origin: nil, type: nil)
   {
     challenge: authenticator_encode(challenge || fake_challenge),
-    origin: origin || fake_origin,
+    origin: origin || "whatever",
     type: type || "webauthn.create"
   }.to_json
 end
```

the test-case validation would fail for both `origin` and `type` (the thing being tested), which is not what we want.

Reply:
I agree. I'll update it 🍻

Reply:
Cool. Thanks!!